examples/camera-app/linux/doc/camera-app-architecture.md

```diff
@@ -0,0 +1,543 @@
+# Camera App Architecture and Design
+
+This document provides a detailed overview of the design and architecture of the
+Matter camera application, focusing on the Linux implementation.
+
+## High-Level Architecture
+
+The camera application is designed with a clear separation between the generic
+Matter cluster logic and the platform-specific hardware abstraction. This is
+achieved through the use of a `CameraDeviceInterface`, which defines the
+contract that any platform-specific camera implementation must adhere to.
+
+The core components are:
+
+-   **Camera App (`CameraApp`):** Responsible for initializing and managing the
+    Matter clusters related to camera functionality. It is platform-agnostic and
+    interacts with the hardware through the `CameraDeviceInterface`.
+-   **Camera Device (`CameraDevice`):** The Linux-specific implementation of the
+    `CameraDeviceInterface`. It manages the camera hardware, using V4L2 and
+    GStreamer, and provides the necessary delegates for the Matter clusters.
+-   **Media Controller (`DefaultMediaController`):** The central hub for media
+    data distribution. It receives encoded media frames from the `CameraDevice`
+    and distributes them to the various transport managers.
+-   **Transport Managers (`WebRTCProviderManager`,
+    `PushAvStreamTransportManager`):** These classes manage the specific
+    transport protocols for streaming media to clients.
+-   **Matter Clusters:** The various Matter clusters that expose the camera's
```

Comment:
should it be Matter Cluster delegates?

Reply:
Will change it to `Matter Cluster Managers`.